pennylane/gradients/finite_difference.py

```diff
@@ -724,13 +724,13 @@ def processing_fn(results):
 
             # compute the linear combination of results and coefficients
             res = qml.math.stack(res)
-            g = sum(c * r for c, r in zip(coeffs, res))
+            g = sum(r * c for c, r in zip(coeffs, res))
 
             if c0 is not None:
                 # add on the unshifted term
-                g = g + c0 * r0
+                g = g + r0 * c0
 
-            grads.append(g / (h**n))
+            grads.append(g * (h ** (-n)))

```

Comment:
This removes the necessity to implement `__truediv__` for the data type, the standard `/` operation between the custom object and a standard dtype like `float`.

Reply:
Nice hack 👍